akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -301,6 +311,22 @@ private[projection] class R2dbcOffsetStore(
       clearInflight()
       if (newState == State.empty) {
         None
+      } else if (recordsWithKey.head.projectionKey != projectionId.key ||
+                 recordsWithKey.iterator
+                   .map(_.projectionKey)
+                   .toSet
+                   .size > 1) { // FIXME more efficient way to find mix of keys
+        // When downscaling projection instances (changing slice distribution) there
+        // is a possibility that one of the previous projection instances was further behind than the backtracking
+        // window, which would cause missed events if we started from latest. In that case we use the latest
+        // offset of the earliest slice
+
+        // FIXME would not be needed if scaling up. To know that we would have to use a convention of keys encoded as
+        // slice range min-max
```

Comment:
This comment is not correct. When scaling up it would have the same projection key for all stored records of the new slice range, but it wouldn't be the same key as the new `projectionId.key`. Above check `recordsWithKey.head.projectionKey != projectionId.key` can be removed. It was mostly an optimization.